subworkflows/local/map.nf

```diff
@@ -19,6 +20,47 @@ workflow MAP {
     ch_versions       = Channel.empty()
     ch_multiqc_files  = Channel.empty()
 
+    if ( params.run_fastq_sharding ) {
+
+        ch_input_for_sharding = reads
+
+        SEQKIT_SPLIT2( ch_input_for_sharding )
+        ch_versions        = ch_versions.mix ( SEQKIT_SPLIT2.out.versions.first() )
+
+        sharded_reads = SEQKIT_SPLIT2.out.reads
+            .transpose()
+            .map {
+                meta, reads ->
+                    new_meta = meta.clone()
+                    new_meta.shard_number = reads.getName().replaceAll(/.*(part_\d+).(?:fastq|fq).gz/, '$1')
+                    [ new_meta, reads ] 
+            }
+            .groupTuple()
```

Comment:
What does the `groupTuple` do here? each shard will have its own unique meta, so there is nothing to group, right? Am I missing something?

Reply:
I forgot why it is needed as well, but i think it had something to do with grouping read pairs when reads were not merged with adapterremoval or fastp

Reply:
oh, that makes sense actually. OK!